front/src/applications/operationalStudies/hooks/usePersistScenarioHeader.ts

```
@@ -0,0 +1,42 @@
+import { useMemo, useState, useEffect } from 'react';
+
+import type { Board } from '../types';
+
+export function usePersistScenarioHeader(scenarioId: number | undefined, defaultBoards: Board[]) {
+  const storageKey = useMemo(
+    () => (scenarioId ? `scenario:${scenarioId}:boards` : `scenario:_pending:boards`),
+    [scenarioId]
+  );
+
+  const [activeBoards, setActiveBoards] = useState<Set<Board>>(() => new Set(defaultBoards));
+
+  useEffect(() => {
+    if (!scenarioId) return;
+    try {
+      const saved = localStorage.getItem(storageKey);
+      if (saved) setActiveBoards(new Set(JSON.parse(saved) as Board[]));
+    } catch {
+      /* ignore */
```

Should we perhaps log the error here?

---

Why do we need a try/catch in the first place? (Same question for try/catch below.)

---

I saw localStorage could throw for some reasons, but I can change that if you think it's safe

---

In some cases a user may have blocked the use of cookies and this prevents the use of localstorage.

---

`localStorage.getItem()` can never fail [per the spec](https://html.spec.whatwg.org/multipage/webstorage.html#dom-storage-getitem).

`localStorage.setItem()` may fail, but we shouldn't silently drop the error in that case.

---

Just to clarify: I think we shouldn't ignore the error, because if a user comes up complaining that the scenario header isn't persisted, debugging the root cause will be tricky. I'm completely fine handling the error and making it non-fatal though, e.g. with

```ts
try {
  localStorage.setItem(…);
} catch (err) {
  console.error('Failed to save scenario boards configuration:', err);
}`